stl/src/syserror_import_lib.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// This must be as small as possible, because its contents are
+// injected into the msvcprt.lib and msvcprtd.lib import libraries.
+// Do not include or define anything else here.
+// In particular, basic_string must not be included here.
+
+#include <__msvc_system_error_abi.hpp>
+
+#include <Windows.h>
+
+namespace {
+
+    struct _Whitespace_bitmap_t {
+        bool _Is_whitespace[256];
+
+        constexpr _Whitespace_bitmap_t() : _Is_whitespace{} {
+            _Is_whitespace[' ']  = true;
+            _Is_whitespace['\n'] = true;
+            _Is_whitespace['\r'] = true;
+            _Is_whitespace['\t'] = true;
+            _Is_whitespace['\0'] = true;
+        }
+
+        constexpr bool _Test(const char _Ch) const noexcept {
+            return _Is_whitespace[static_cast<unsigned char>(_Ch)];
+        }
+    };
+
+    constexpr _Whitespace_bitmap_t _Whitespace_bitmap;
+
+} // unnamed namespace
+
+_EXTERN_C
+
+_NODISCARD size_t __CLRCALL_PURE_OR_STDCALL __std_get_string_size_without_trailing_whitespace(
+    _In_ const char* const _Str, _In_ size_t _Size) noexcept {
+    for (;;) {
+        if (_Size == 0) {
+            return _Size;
+        }
+
+        --_Size;
+        if (!_Whitespace_bitmap._Test(_Str[_Size])) {
+            ++_Size;
+            return _Size;
+        }
+    }

```

Comment:
I believe that this control flow can be significantly simplified (warning: not tested):
```cpp
while (_Size != 0 && _Whitespace_bitmap._Test(_Str[_Size - 1])) {
    --_Size;
}
return _Size;
```

This isn't a classic for-loop so I recommend resisting the temptation to make it a for-loop with an empty body. However, I believe that a while-loop is appropriate - we can see the invariants more clearly this way, and we avoid the mental complexity of decrementing and incrementing size when a non-whitespace character is found. (There is no efficiency concern around saying `_Size - 1` followed by decrementing `_Size`.)

Reply:
That was indeed attempting to avoid repeated subtraction of _Size. I'll see what codegen looks like.

Reply:
My heuristic is that simplifying control flow for humans to understand is worth 1000x avoiding an integer operation in debug mode. (In release mode I would be astounded to see two decrements.)

Reply:
Right, I'm going to check release mode. I agree debug mode difference is immaterial.

Reply:
Seems OK https://gcc.godbolt.org/z/iZFdGd